spec/models/user_spec.rb

```diff
@@ -0,0 +1,67 @@
+require "rails_helper"
+
+RSpec.describe User, type: :model do
+  subject {FactoryBot.create :user}
+
+  describe ".create" do
+    it {is_expected.to be_valid}
+  end
+
+  context "validations" do
+    describe "name valid" do
+      it {is_expected.to validate_presence_of :name}
+      it {is_expected.to validate_length_of(:name).is_at_most Settings.validates.max_name}
+    end
+
+    describe "email valid" do
+      it {is_expected.to validate_presence_of :email}
+      it {is_expected.to validate_length_of(:email).is_at_most Settings.validates.max_email}
+      it {expect(subject.email).to match(Settings.validates.valid_email) }
+      it {is_expected.to validate_uniqueness_of(:email).case_insensitive}
+    end
+
+    describe "address valid" do
+      it {is_expected.to validate_presence_of :address}
+      it {is_expected.to validate_length_of(:address).is_at_most Settings.validates.max_address}
+    end
+
+    describe "gender valid" do
+      it {is_expected.to validate_presence_of :gender}
+    end
+
+    describe "birthday valid" do
+      it {is_expected.to validate_presence_of :birthday}
+    end
+
+    describe "company valid" do
+      it {should validate_presence_of(:company)}
+      it {is_expected.to validate_length_of(:company).is_at_most Settings.validates.max_company}
+    end
+  end
+
+  context "associations" do
+    it {is_expected.to have_many(:images).dependent :destroy}
+    it {should have_many(:active_messages).class_name(Message.name).with_foreign_key("sender_id").dependent :destroy}
```

Comment:
Hạn chế dùng should nhé em, chuyển thành is_expected.to hết đi

Reply:
Dạ anh